cubi_tk/irods/check.py

```diff
@@ -84,87 +111,104 @@ def execute(self):
         res = self.check_args(self.args)
         if res:  # pragma: nocover
             return res
-
         logger.info("Starting cubi-tk irods %s", self.command_name)
-        logger.info("  args: %s", self.args)
-
-        self.run_tests(self.get_files())
+        logger.info("Args: %s", self.args)
+
+        # Check for environment file
+        if not os.path.isfile(self.irods_env_path):
+            logger.error("iRODS environment not found in %s", self.irods_env_path)
+            raise FileNotFoundError
+        with open(self.irods_env_path, "r") as f:
+            irods_env = json.load(f)
+        logger.info("iRODS environment: %s", irods_env)
+
+        # Connect to iRODS
+        irods = self.connect()
+        try:
+            root_coll = irods.collections.get(self.args.irods_path)
+            logger.info("iRODS connection initialized")
+        except Exception as e:
+            logger.error("Failed to retrieve iRODS path: %s", self.get_irods_error(e))
+            raise
 
+        # Get files and run checks
+        logger.info("Querying for data objects")
+        data_objs = self.get_data_objs(root_coll)
+        self.run_checks(irods, data_objs)
         logger.info("All done")
-        return None
 
-    def run_tests(self, files):
-        """Run tests in parallel."""
-        num_files = len(files["files"])
-        lst_files = "\n".join(files["files"][:19])
-        logger.info("Checking %s files (first 20 shown):\n%s", num_files, lst_files)
+    def run_checks(self, irods: iRODSSession, data_objs: dict):
+        """Run checks on files, in parallel if enabled."""
+        num_files = len(data_objs["files"])
+        lst_files = "\n".join([f.path for f in data_objs["files"]][:19])
+        logger.info(
+            "Checking %s file%s%s:\n%s",
+            num_files,
+            "s" if num_files != 1 else "",
+            " (first 20 shown)" if num_files > 20 else "",
```

Comment:
Can we make the `20` configurable and have `0` mean "no limit"?

Reply:
Why not. That part wasn't done by me to begin with :)

Reply:
Done.